test_util/src/lsp.rs

```diff
@@ -148,13 +162,42 @@ impl LspClient {
   where
     R: de::DeserializeOwned,
   {
+    if !self.msg_queue.is_empty() {
+      let mut msg_queue = VecDeque::new();
+      loop {
+        match self.msg_queue.pop_front() {
+          Some(LspMessage::Notification(method, maybe_params)) => {
+            if let Some(p) = maybe_params {
+              let params = serde_json::from_value(p)?;
+              return Ok((method, Some(params)));
+            } else {
+              return Ok((method, None));
+            }
```

Comment:
Can this be simplified to  
```rust
return Ok((method, maybe_params.try_into()?));
```

Reply:
Something non-obvious is going on with that suggestion... I think it is because of the generic type:

```
the trait bound `std::option::Option<R>: From<std::option::Option<Value>>` is not satisfied
required because of the requirements on the impl of `Into<std::option::Option<R>>` for `std::option::Option<Value>`
required because of the requirements on the impl of `TryFrom<std::option::Option<Value>>` for `std::option::Option<R>`
required because of the requirements on the impl of `TryInto<std::option::Option<R>>` for `std::option::Option<Value>`
```

If it where a concrete type, I suspect it would work.  `.into()` doesn't work here either, and even unwrapping the value and using `.try_into()` or `.into()` doesn't work.

I have simplified it by one line (since there is no `.try_map()` that is part of the Rust std library):

```rust
let maybe_params = match maybe_params {
  Some(params) => Some(serde_json::from_value(params)?),
  None => None
};
return Ok((method, maybe_params));
```